core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -777,6 +787,73 @@ NavigableMap<Long, InFlightBatch> cachedState() {
         return new ConcurrentSkipListMap<>(cachedState);
     }
 
+    // TODO: maxDeliveryCount should be utilized here once it is implemented
+    /**
+     * Apply acquisition lock to acquired records.
+     * @param memberId The member id of the client that is putting the acquisition lock.
+     * @param baseOffset The base offset of the acquired records.
+     * @param lastOffset The last offset of the acquired records.
+     * @param cachedStateEntry The cached state map entry for the acquired records which contains the entire batch
+     *                         from base offset to last offset.
+     */
+    public void applyAcquisitionLock(String memberId, long baseOffset, long lastOffset, Map.Entry<Long, InFlightBatch> cachedStateEntry) {
```

Comment:
I'm not sure `applyAcquisitionLock` is a good method name. If I understand correctly, it's really `releaseAcquisitionLock` based on the comments "unable to release" below. It's a bit confusing right now.

Reply:
the `run()` inside `applyAcquisitionLock()` gets triggered when the lock times out, hence you see the comments of "unable to release" inside it. However, we call `applyAcquisitionLock()` when we are acquiring records. So I understand where the confusion is coming from. I can think of 2 ways how to resolve this naming issue -
1. Add the comment above `run()` function to describe that it runs once the timeout happens.
2. Rename `applyAcquisitionLock()` to `acquisitionLock()` since it is a complete functionality.

Reply:
Ah, of course. It's obvious when you say it. Maybe `startAcquisitionLockTimer` and the comments on the `run()` method.

Reply:
I would have divided this method in 2:
1. Move the TimerTask to separate method where we return the TimerTask itself.
2. Change method name to `scheduleLockTimeout`